From: Christian Marangi <ansuelsmth@gmail.com>
Date: Thu, 23 Oct 2025 16:58:48 +0200
Subject: [PATCH] net: airoha: use device_set_node helper to setup GDM node

Use device_set_node helper to setup GDM node instead of manually setting
the dev.of_node to also fill the fwnode entry.

This is to address some API that use fwnode instead of of_node (for
example phylink_create).

Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>
---
 drivers/net/ethernet/airoha/airoha_eth.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/net/ethernet/airoha/airoha_eth.c
+++ b/drivers/net/ethernet/airoha/airoha_eth.c
@@ -4437,7 +4437,7 @@ static int airoha_alloc_gdm_device(struc
 		return -EINVAL;
 	}
 
-	netdev->dev.of_node = of_node_get(np);
+	device_set_node(&netdev->dev, of_fwnode_handle(of_node_get(np)));
 	dev = netdev_priv(netdev);
 	u64_stats_init(&dev->stats.syncp);
 	dev->port = port;
